Runtime/Editor/VisualElements/LootLocker MainWindow/LootLockerMainWindow.cs

```diff
@@ -0,0 +1,503 @@
+        keyList.AddToClassList("keylist");
+
+        keyEnvironment = root.Q<DropdownField>("keyEnvironment");
+        gameEnvironment = root.Q<DropdownField>("gameEnvironment");
+
+        keyName = root.Q<TextField>("apikeyname");
+
+        createKeyButton = root.Q<Button>("create");
+
+        createKeyButton.clickable.clicked += CreateKey;
+
+        sheet = AssetDatabase.LoadAssetAtPath<StyleSheet>("Assets/LootLockerSDK/Runtime/Editor/VisualElements/LootLocker MainWindow/LootLockerMainWindow.uss");
+        root.styleSheets.Add(sheet);
+    }
+
+
+
+    private void Update()
+    {
+        returnBtn.style.display = currentPage == Page.OrganisationPage ? DisplayStyle.None : DisplayStyle.Flex;
+
+    }
+
+    void Return(EventBase e)
+    {
+        if (e.propagationPhase != PropagationPhase.AtTarget)
+            return;
+        var target = e.target as Button;
+
+        switch (currentPage)
+        {
+            case Page.GamePage:
+                optionList.Clear();
+                PopulateList(ContentType.Organisations);
+                break;
+            case Page.GameOptionPage:
+                optionList.Clear();
+                PopulateList(ContentType.Games);
+                optionList.style.display = DisplayStyle.Flex;
+                tabWindow.style.display = DisplayStyle.None;
+                break;
+            case Page.ApiPage:
+                optionList.Clear();
+                PopulateList(ContentType.Games);
+                gameTitle.text = "";
+                optionList.style.display = DisplayStyle.Flex;
+                tabWindow.style.display = DisplayStyle.None;
+                activeWindowLabel.style.display = DisplayStyle.None;
+                break;
+            default:
+                target.style.display = DisplayStyle.None;
+                break;
+        }
+
+    }
+
+    private void OpenAPIKeyTab()
+    {
+        keyList.Clear();
+
+        tabWindow.style.display = DisplayStyle.Flex;
+        currentPage = Page.ApiPage;
+
+        activeWindowLabel.style.display = DisplayStyle.Flex;
+        activeWindowLabel.text = " - API Keys";
+
+        EditorApplication.update += OnEditorUpdate;
+        LootLockerAdminManager.GetAllKeys(activeGame.id.ToString(), (onComplete) =>
+        {
+            if (onComplete.success)
+            {
+                foreach (var key in onComplete.api_keys)
+                {
+                    CreateAPIKeyTemplate(key);
+                }
+            }
+            EditorApplication.update -= OnEditorUpdate;
+        });
+    }
+
+    public void CreateKey()
+    {
+        int gameEnv = activeGame.id;
+
+        if (gameEnvironment.value == "Stage")
+        {
+            gameEnv = activeGame.development.id;
+        }
+
+        EditorApplication.update += OnEditorUpdate;
+        
+        LootLockerAdminManager.GenerateKey(gameEnv.ToString(), keyName.value, keyEnvironment.value.ToLower(), (onComplete) =>
```

Comment:
We should probably sanity check the input?

Reply:
Checking if its empty, next iteration will also give a popup